plugins/chat/assets/javascripts/discourse/services/chat-state-manager.js

```diff
@@ -68,6 +80,21 @@ export default class ChatStateManager extends Service {
   }
 
   didCloseDrawer() {
+    withPluginApi("1.8.0", (api) => {
+      api.setSidebarPanel("main");
+
+      if (getUserChatSeparateSidebarMode(this.currentUser).fullscreen) {
```

Comment:
The same as above, we can at minimum have a variable, but better a private property.

Reply:
It's confusing that we have three copies of this code:

```
if (mode.fullscreen) then ...
else if (mode.always) then...
else

```

and every copy is a bit different from others copies. For example, this copy is the only one where we call `api.showSidebarSwitchPanelButtons();` in the else-if branch. Is that intentional?

Would be great to dry it up and have only one place where we do it, or probably two places - one for setting combined / separated mode and one for showing / hiding buttons.

Reply:
> and every copy is a bit different from others copies. For example, this copy is the only one where we call api.showSidebarSwitchPanelButtons(); in the else-if branch. Is that intentional?

what do you mean?

Reply:
OK I got it, you mean compared to the one in routes

Reply:
I removed it I don't think we need this anymore, it was something present in kris example, but I don't think I need it actually :+1:

Reply:
Yep, to the one in routes. And there is also another one in initializers/chat-sidebar.js

Reply:
Yes I removed the one in sidebar

Reply:
Not sure, maybe you haven't pushed everything yet, but still there are two copies of this if-else:

https://github.com/discourse/discourse/blob/7ca178f99e4095dde8bde28e9939c1ab18d24c5f/plugins/chat/assets/javascripts/discourse/services/chat-state-manager.js#L86-L95

and:

https://github.com/discourse/discourse/blob/7ca178f99e4095dde8bde28e9939c1ab18d24c5f/plugins/chat/assets/javascripts/discourse/routes/chat.js#L86-L97